streaming/stream_transfer_task.cc

```diff
@@ -70,7 +70,7 @@ struct send_info {
         , cf(tbl_)
         , ranges(std::move(ranges_))
         , prs(dht::to_partition_ranges(ranges))
-        , reader(cf.make_streaming_reader(cf.schema(), std::move(permit_), prs, {}))
+        , reader(cf.make_streaming_reader(cf.schema(), std::move(permit_), prs, gc_clock::now()))
         , update(std::move(update_fn))
```

Comment:
If the clocks on the other node have a significant lag, they'll observe different expiration times.

Concrete example:
 - this node has the clocks set in the future
 - other nodes have correct clocks
 - we stream to the other nodes
 - other nodes observe premature expiration

but, it's not like this patch is making anything worse; compaction could have kicked a little while ago and expired that data anyway. So no problem here.

Reply:
This was my thinking as well.